api/pipeline_spec.proto

```diff
@@ -0,0 +1,420 @@
+  // ```
+  // tasks:
+  // - task_info:
+  //     name: trainer
+  //   executor_label: trainer
+  // deployment_config:
+  //   @type: cloud.ml.pipelines.v1alpha3.proto.PipelineDeploymentConfig
+  //   executors:
+  //     trainer:
+  //       container:
+  //         image: gcr.io/tfx:latest
+  //         args: []
+  // ```
+  string executor_label = 4;
+
+  // A list of names of upstream tasks that do not provide input
+  // artifacts for this task, but nonetheless whose completion this task depends
+  // on.
+  repeated string dependent_tasks = 5;
+
+  message CachingOptions {
+    // Whether or not to enable cache for this task. Defaults to false.
+    bool enable_cache = 1;
+  }
+  CachingOptions caching_options = 6;
+}
+
+// Basic info of a pipeline.
+message PipelineInfo {
+  // Required field. The name of the pipeline.
+  // The name will be used to create or find pipeline context in MLMD.
+  string name = 1;
+}
+
+// The definition of a artifact type in MLMD.
+message ArtifactTypeSchema {
+  oneof kind {
+    // The name of the type. The format of the title must be:
+    // `<namespace>.<title>.<version>`.
+    // Examples:
+    //  - `aiplatform.Model.v1`
+    //  - `acme.CustomModel.v2`
+    // When this field is set, the type must be pre-registered in the MLMD
+    // store.
+    string schema_title = 1;
+
+    // Points to a YAML file stored on Google Cloud Storage describing the
+    // format.
+    string schema_uri = 2;
+
+    // Contains a raw YAML string, describing the format of
+    // the properties of the type.
+    string instance_schema = 3;
+  }
+}
+
+// The basic info of a task.
+message PipelineTaskInfo {
+  // The unique name of the task within the pipeline definition. This name
+  // will be used in downstream tasks to indicate task and data dependencies.
+  string name = 1;
+}
+
+// Definition for a value or reference to a runtime parameter. A
+// ValueOrRuntimeParameter instance can be either a field value that is
+// determined during compilation time, or a runtime parameter which will be
+// determined during runtime.
+message ValueOrRuntimeParameter {
+  oneof value {
+    // Constant value which is determined in compile time.
+    Value constant_value = 1;
+    // Name of the runtime parameter.
+    string runtime_parameter = 2;
+  }
+}
+
+// The definition of the deployment config of the pipeline. It contains the
+// the platform specific executor configs for KFP OSS.
+message PipelineDeploymentConfig {
+  // The specification on a container invocation.
+  // The string fields of the message support string based placeholder contract
+  // defined in [ExecutorInput](). The output of the container follows the
+  // contract of [ExecutorOutput]().
+  message PipelineContainerSpec {
+    // The image uri of the container.
+    string image = 1;
+    // The main entrypoint commands of the container to run. If not provided,
+    // fallback to use the entry point command defined in the container image.

```

Comment:
Can we make this required?

Reply:
Why? it's not required for both TFX and KFP today. Making it required will break existing component behavior.